File: cadwiki-nuget/cadwiki.AdminUtils/Form1.vb
Imports System.IO
Imports System.Reflection
Imports System.Text.RegularExpressions
Imports cadwiki.NetUtils

Public Class Form1

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim asm As Assembly = Assembly.GetExecutingAssembly()
        Dim assemblyName = asm.GetName()
        Dim version As Version = assemblyName.Version
        Dim versionString As String = version.ToString()
        LabelCurrentVersion.Text = versionString
        Dim newVersion As Version = version
        newVersion = newVersion.IncrementBuild
        Dim time As DateTime = DateTime.Now
        Dim format As String = "HHmm"
        Dim timeStamp As String = time.ToString(format)
        newVersion = newVersion.SetVersion(newVersion.Major, newVersion.Minor, newVersion.Build, CInt(timeStamp))
        TextBoxNewVersion.Text = newVersion.ToString()

    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim asm As Assembly = Assembly.GetExecutingAssembly()
        Dim dllPath As String = asm.Location
        Dim dllFolder As String = Path.GetDirectoryName(dllPath)
        Dim root As String = Paths.TryGetSolutionDirectoryPath(dllFolder)
        Dim repoFolder As String = New DirectoryInfo(root).Parent.FullName
        Dim folders As List(Of String) =
            System.IO.Directory.GetDirectories(repoFolder, "*", System.IO.SearchOption.AllDirectories).ToList()
        folders.Reverse()
        folders.Add(repoFolder)
        folders.Reverse()
        Dim projectsToUpdate As New List(Of String) From {
            "*cadwiki-nuget",
            "*CadDevToolsDriver*",
            "*cadwiki.AdminUtils*",
            "*cadwiki.CadDevTools*",
            "*cadwiki.DllReloader*",
            "*cadwiki.FileStore*",
            "*cadwiki.NetUtils*",
            "*cadwiki.NUnitTestRunner*",
            "*cadwiki.WpfUi*"
        }
        Dim wildCardPatterns As New List(Of String) From {
             "*apikeys.txt",
            "*README.nuget.commands.md",
            "*AssemblyInfo.vb",
            "*.nuspec",
            "*.targets"
        }
        For Each folder In folders
            Dim folderName As String = Path.GetFileName(folder)
            Dim isThisAProjectToUpdate As Boolean = False
            For Each project In projectsToUpdate
                If StringContains(project, folder) Then
                    isThisAProjectToUpdate = True
                    Exit For
                End If
            Next

            If (isThisAProjectToUpdate) Then
                For Each fileName In Directory.GetFiles(folder)
                    For Each wildCardPattern In wildCardPatterns
                        If StringContains(wildCardPattern, fileName) Then
                            ReplaceText(LabelCurrentVersion.Text,
                                        TextBoxNewVersion.Text, fileName)
                        End If
                    Next
                Next
            End If

        Next

    End Sub

    Private Sub ReplaceText(oldText As String, newText As String, fileName As String)
        Dim lines As List(Of String) = IO.File.ReadAllLines(fileName).ToList
        For index As Integer = 0 To lines.Count - 1
            If lines(index).Contains(oldText) Then
                lines(index) = lines(index).Replace(oldText, newText)
            End If
        Next
        IO.File.WriteAllLines(fileName, lines.ToArray)
    End Sub

    Public Function StringContains(ByVal searchPattern As String,
                             ByVal inputText As String) As Boolean
        Dim regexText As String = WildcardToRegex(searchPattern)
        Dim regex As Regex = New Regex(regexText, RegexOptions.IgnoreCase)

        If regex.IsMatch(inputText) Then
            Return True
        End If
        Return False
    End Function

    Public Shared Function WildcardToRegex(ByVal pattern As String) As String
        Return "^" & Regex.Escape(pattern).Replace("\*", ".*").Replace("\?", ".") & "$"
    End Function

End Class

File: cadwiki-nuget/cadwiki.AdminOps/Form1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Text.RegularExpressions;
using cadwiki.NetUtils;

namespace cadwiki.AdminOps
{

    public partial class Form1
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            var asm = Assembly.GetExecutingAssembly();
            var assemblyName = asm.GetName();
            var version = assemblyName.Version;
            string versionString = version.ToString();
            LabelCurrentVersion.Text = versionString;
            var newVersion = version;
            newVersion = newVersion.IncrementBuild();
            var time = DateTime.Now;
            string format = "HHmm";
            string timeStamp = time.ToString(format);
            newVersion = newVersion.SetVersion(newVersion.Major, newVersion.Minor, newVersion.Build, int.Parse(timeStamp));
            TextBoxNewVersion.Text = newVersion.ToString();

        }

        private void Button1_Click(object sender, EventArgs e)
        {
            var asm = Assembly.GetExecutingAssembly();
            string dllPath = asm.Location;
            string dllFolder = Path.GetDirectoryName(dllPath);
            string root = Paths.TryGetSolutionDirectoryPath(dllFolder);
            string repoFolder = new DirectoryInfo(root).Parent.FullName;
            var folders = Directory.GetDirectories(repoFolder, "*", SearchOption.AllDirectories).ToList();
            folders.Reverse();
            folders.Add(repoFolder);
            folders.Reverse();
            var projectsToUpdate = new List<string>() { "*cadwiki-nuget", "*CadDevToolsDriver*", "*cadwiki.AdminUtils*", "*cadwiki.CadDevTools*", "*cadwiki.DllReloader*", "*cadwiki.FileStore*", "*cadwiki.NetUtils*", "*cadwiki.NUnitTestRunner*", "*cadwiki.WpfUi*" };
            var wildCardPatterns = new List<string>() { "*apikeys.txt", "*README.nuget.commands.md", "*AssemblyInfo.vb", "*.nuspec", "*.targets" };
            foreach (var folder in folders)
            {
                string folderName = Path.GetFileName(folder);
                bool isThisAProjectToUpdate = false;
                foreach (var project in projectsToUpdate)
                {
                    if (StringContains(project, folder))
                    {
                        isThisAProjectToUpdate = true;
                        break;
                    }
                }

                if (isThisAProjectToUpdate)
                {
                    foreach (var fileName in Directory.GetFiles(folder))
                    {
                        foreach (var wildCardPattern in wildCardPatterns)
                        {
                            if (StringContains(wildCardPattern, fileName))
                            {
                                ReplaceText(LabelCurrentVersion.Text, TextBoxNewVersion.Text, fileName);
                            }
                        }
                    }
                }

            }

        }

        private void ReplaceText(string oldText, string newText, string fileName)
        {
            var lines = File.ReadAllLines(fileName).ToList();
            for (int index = 0, loopTo = lines.Count - 1; index <= loopTo; index++)
            {
                if (lines[index].Contains(oldText))
                {
                    lines[index] = lines[index].Replace(oldText, newText);
                }
            }
            File.WriteAllLines(fileName, lines.ToArray());
        }

        public bool StringContains(string searchPattern, string inputText)
        {
            string regexText = WildcardToRegex(searchPattern);
            var regex = new Regex(regexText, RegexOptions.IgnoreCase);

            if (regex.IsMatch(inputText))
            {
                return true;
            }
            return false;
        }

        public static string WildcardToRegex(string pattern)
        {
            return "^" + Regex.Escape(pattern).Replace(@"\*", ".*").Replace(@"\?", ".") + "$";
        }

    }
}